car_cleaning_service_with_threads.rb

```diff
@@ -0,0 +1,115 @@
+require'time'
+
+class Car
+  attr_reader :id
+
+  def initialize(id)
+    @id = id
+  end
+
+  def to_s
+    @id
+  end
+
+  def to_sym
+    @id.to_sym
+  end
+end
+
+class CarCleaningService
+
+  MAX_CARS_AT_ONCE = 2
+  CAR_CLEANING_TIME = 2
+  ONE_HOUR_IN_SECONDS = 3600
+
+  attr_reader :time
+
+  def initialize(queue = Queue.new)
+    @queue = queue
+    @ready = {}
+    @processed_cars = 0
+    @time = Time.parse(Time.now.strftime("%Y-%m-%dT08:00:00%z"))
+    self.init_threads
+  end
+
+  public
```

Comment:
why do you need to specify `public` here?

Reply:
I only specified it for easier finding of the public methods, but I have no need for it as it is implied here.